boot/bootutil/src/bootutil_priv.h

```diff
@@ -58,10 +58,16 @@ struct flash_area;
      defined(MCUBOOT_SWAP_USING_MOVE) + \
      defined(MCUBOOT_DIRECT_XIP) + \
      defined(MCUBOOT_RAM_LOAD) + \
-     defined(MCUBOOT_FIRMWARE_LOADER)) > 1
+     defined(MCUBOOT_FIRMWARE_LOADER) + \
+     defined(MCUBOOT_SWAP_USING_SCRATCH)) > 1
 #error "Please enable only one of MCUBOOT_OVERWRITE_ONLY, MCUBOOT_SWAP_USING_MOVE, MCUBOOT_DIRECT_XIP, MCUBOOT_RAM_LOAD or MCUBOOT_FIRMWARE_LOADER"
 #endif
 
+#if !defined(MCUBOOT_DIRECT_XIP) && \
+     defined(MCUBOOT_DIRECT_XIP_REVERT)
+#error "MCUBOOT_DIRECT_XIP_REVERT cannot be enabled unless MCUBOOT_DIRECT_XIP is used"
+#endif
+
```

Comment:
I'm not sure it's needed. `MCUBOOT_DIRECT_XIP_REVERT` is a sub-option of `MCUBOOT_DIRECT_XIP` and it has no effect without the direct XIP strategy, therefore it's safe to abandon this check. 
On the other hand it can draw attention to a potential misconfiguration.